bundle/src/main/java/com/adobe/acs/commons/workflow/synthetic/impl/SyntheticWorkflowRunnerImpl.java

```diff
@@ -184,33 +205,40 @@ private void run(final ResourceResolver resourceResolver,
             }
 
             if (workflowProcess != null) {
-
-                // Each Workflow Process Step gets its own workItem whose life starts and ends w the WF Process
-                final SyntheticWorkItem workItem = new SyntheticWorkItem(workflowData);
-                workItem.setWorkflow(workflow);
-
-                // Used to pass in per-Step parameters
-                final MetaDataMap workflowProcessMetaDataMap =
-                        new SyntheticMetaDataMap(metaDataMaps.get(workflowProcessId));
-
                 final long start = System.currentTimeMillis();
-                log.trace("Executing synthetic workflow process [ {} ] on [ {} ]", workflowProcessId, payloadPath);
 
                 try {
-                    // Execute the Workflow Process
-                    workflowProcess.execute(workItem, workflowSession, workflowProcessMetaDataMap);
-                    workItem.setTimeEnded(new Date());
-
-                } catch (SyntheticCompleteWorkflowException ex) {
-                    // Workitem force-completed via a call to workflowSession.complete(..)
-                    workItem.setTimeEnded(new Date());
-                    log.trace(ex.getMessage());
-
+                    final SyntheticMetaDataMap workflowProcessMetaDataMap = new SyntheticMetaDataMap(metaDataMaps.get(workflowProcessId));
+
+                    if (SyntheticWorkflowProcess.Type.GRANITE.equals(workflowProcess.getWorkflowType())) {
+                        runGraniteWorkflowProcess(session, graniteWorkflow, workflowProcessMetaDataMap, workflowProcess);
+                    } else if (SyntheticWorkflowProcess.Type.CQ.equals(workflowProcess.getWorkflowType())) {
+                        runCqWorkflowProcess(session, cqWorkflow, workflowProcessMetaDataMap, workflowProcess);
+                    } else {
+                        log.warn("Workflow process step is of an unknown type [ {} ]. Skipping.", workflowProcess.getWorkflowType());
+                    }
                 } catch (SyntheticTerminateWorkflowException ex) {
                     // Terminate entire Workflow execution for this payload
                     terminated = true;
-                    log.info("Synthetic workflow execution stopped via terminate for [ {} ]", payloadPath);
+                    log.info("Synthetic CQ workflow execution stopped via terminate for [ {} ]", payloadPath);
+                    break;
+                } catch (com.adobe.acs.commons.workflow.synthetic.impl.granite.exceptions.SyntheticTerminateWorkflowException ex) {
```

Comment:
Will leave for now as it can help w debugging (more so if someone leaves an issue for us) the CQ vs Granite distinctions help track down which impl channel to look to, which can otherwise get confusing.